javascript/feed.mjs

```diff
@@ -0,0 +1,149 @@
+import { API_URL } from "./functions.mjs";
+import { showPosts } from "./functions.mjs";
+import { createPost } from "./functions.mjs";
+import { loadJWT } from "./functions.mjs";
+
+const feedForm = document.querySelector("#feedForm");
+const feedContainer = document.querySelector(".feedContainer");
+feedContainer.innerHTML = "";
+loadJWT("token");
+loadJWT("user");
+
+const searchForm = document.querySelector(".searchForm");
+
+searchForm.addEventListener("submit", async (e) => {
+  e.preventDefault();
+  await showPosts();
+  location.href = `/search/index.html?id=${searchForm.search.value}`;
+});
+
+feedForm.addEventListener("submit", async (e) => {
+  e.preventDefault();
+  const form = e.target;
+  const formData = new FormData(form);
+  const post = Object.fromEntries(formData.entries()); //stolen line from Oliver
+  createPost(post);
+  feedContainer.innerHTML = "";
+  await showAllThePostsOnThePage();
+});
+
+/**
+ *
+ * @param {array} postData array of objects from the API
+ * @returns takes data from the API and returns all the posts rendered to HTML
+ */
+function renderPosts(postData) {
+  //transforms the date to a more readable date
+  let strDate = postData.created.split("T")[0];
+
+  if (postData.media) {
```

Comment:
You could have been a bit more terse here by embedding the if clause within your template:

```js
const example = `<div>
${postData.media ? `<img src="${postData.media}"/>` : "No image"}
</div>
`
```